### PR TITLE
Monkey mutations

### PR DESCRIPTION
**Pull request by chloecat0**
## About The Pull Request
Makes it so any mob with dna is able to use mutations rather than just humans, right now the only other mobs which have dna are monkeys. Also generally improves genetics code and fixes some runtimes I found while testing.

## Why It's Good For The Game
Mutations are stored in the dna so it's a bit weird that only humans can use them, almost all mutations had no reason to be on humans rather than carbons in general so this helps simplify things a bit and makes it much easier to add more mobs which can have mutations in future.

## Testing Photographs and Procedure

<details>

peak_performance

<summary>Screenshots&Videos</summary>
</details>

## Changelog
:cl:
add: Anything with dna can get mutations (such as monkeys)
code: Added mobtypes_allowed var to mutations
code: Improvements to genetics code
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
what is a dwarfism monke? just shortened sprited monke?

**Comment by chloecat0** [comment]
> what is a dwarfism monke? just shortened sprited monke?

Yeah it just scales them down slightly like in humans

**Comment by imsynthex** [comment]
can't wait to have a hulk monkey ventcrawl into my workplace and kill me instantly

i have some balance concerns, but the idea is good

**Comment by zeskorion** [comment]
> can't wait to have a hulk monkey ventcrawl into my workplace and kill me instantly
> 
> i have some balance concerns, but the idea is good

Hulk is human only

File: code/_onclick/other_mobs.dm [hunk 1]
```diff
@@ -115,7 +117,13 @@
 /*
 	Monkeys
 */
-/mob/living/carbon/monkey/UnarmedAttack(atom/A)
+/mob/living/carbon/monkey/UnarmedAttack(atom/A, proximity)
+	var/override = 0
+	for(var/datum/mutation/HM as() in dna.mutations)
+		override += HM.on_attack_hand(A, proximity)
+	if(override)
+		return
```
Comment (PowerfulBacon) [inline code/_onclick/other_mobs.dm:125]: What is the point of override?
Returning at the end of a proc will do nothing, so this is essentially the same as
```
/mob/living/carbon/monkey/UnarmedAttack(atom/A, proximity)
	for(var/datum/mutation/HM as() in dna.mutations)
		HM.on_attack_hand(A, proximity)
```

**Comment by PowerfulBacon** [comment]
Missed this PR

File: code/_onclick/other_mobs.dm (hunk 1 above)
Comment (chloecat0) [reply code/_onclick/other_mobs.dm:125]: Override stops attack_paw from being called on line 127

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
yolo

**Pull request merged**